src/main/java/nextstep/subway/station/domain/Station.java

```diff
@@ -26,4 +33,17 @@ public Long getId() {
     public String getName() {
         return name;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Station station = (Station) o;
+        return Objects.equals(id, station.id);
+    }
```

Comment:
```suggestion
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Station)) return false;
        Station station = (Station) o;
        return Objects.equals(getId(), station.getId());
    }
```
우선 프록시 객체가 어떨때 생기는지 찾아보세요!
또한 entity 의 equals는 조심히 사용하셔서 프록시 객체를 비교하게 되더라도 올바르게 동작할 수 있도록 해주어야 합니다. :)
이번 리뷰는 수정하신 후에 진행하도록 할게요!

Reply:
제안해주신 방법으로 equals를 수정하니 테스트 케이스가 바로 성공되네요!! 🥰
며칠 동안 고민하면서 답답한 마음에 PR로 질문드렸는데 명쾌한 해결 방법을 알려주셔서 정말 감사드립니다. 👍

우선, 지하철 노선 조회 API 요청 시 proxy 객체가 생성된 이유는 `Line` entity 클래스 멤버 변수로 정의된,
`Sections` 일급 컬렉션에 sections가 `fetch = FetchType.LAZY`로 되어있어서 proxy 객체가 생성되는 것으로 알고 있습니다.

기존 로직에서는 proxy 객체가 `if (o == null || getClass() != o.getClass()) return false;` 로직을 타면서,
proxy 객체 비교 결과가 false로 반환되어 발생한 문제로 파악되었습니다.

따라서, equals를 `if (!(o instanceof Station)) return false;`로 수정해서 Station proxy 객체도 같은 타입으로 판단하고,
`return Objects.equals(getId(), station.getId());` 구문에서 값을 비교하도록 가이드해주신 부분 확인했습니다. 🙆‍♂️

다만, 답변해주신 내용에 대해 좀 더 아티클들을 찾아보면서 질문이 한 가지 생겼습니다.
entity 클래스는 실제 객체가 아닌 proxy 객체를 반환할 수도 있어서 entity 클래스의 equals 사용을 지양하는 것이 좋을까요?

만약, entity 클래스 equals 사용을 지양해야 한다면 컬렉션을 사용하는 이점을 누리지 못하게 될 것 같은데,
`Objects.equals(getId(), station.getId());` 형태로 유니크한 값을 직접 비교하는 방식이 지향될까요? 🤔

p.s.
이번 미션에서 리뷰어님 덕분에 많은 것을 배울 수 있었는데,
답답했던 부분도 한 번에 해결해주셔서 다시 한번 감사드립니다!! 😍

Reply:
equals는 다양한 곳에서 사용되기 때문에 equals 사용을 지양하는 것은 옳지 않은 결론인거 같아요
말씀드렷듯이 entity 클래스의 equals를 프록시 객체와 비교시에도 잘 동작하도록 오버라이드 하신 후에 사용해야합니다.

Reply:
entity 클래스에서 `equals`는 proxy 객체를 생각하며 항상 주의해서 사용해야겠네요. 😉
의견 감사드립니다!